lib/openstudio/extended_runner.rb

```diff
@@ -0,0 +1,64 @@
+# Extend OS Runner to persist measure information throughout the workflow
+class ExtendedRunner < OpenStudio::Ruleset::OSRunner
+  # Allow former arguments to be set and read
+  # TODO: Consider having the former arguments passed in in initialization, and define as attr_reader
+  attr_accessor :former_workflow_arguments
+  attr_reader :workflow_arguments
+
+  # Add in @workflow_arguments
+  def initialize
+    super
+
+    @workflow_arguments = nil
+  end
+
+  # Take the OS Argument type and map it correctly to the argument value.
+  #     OPENSTUDIO_ENUM( OSArgumentType,
+  #     ((Boolean)(Bool)(0))
+  #     ((Double)(Double)(1))
+  #     ((Quantity)(Quantity)(2))
+  #     ((Integer)(Int)(3))
+  #     ((String)(String)(4))
+  #     ((Choice)(Choice)(5))
+  #     ((Path)(Path)(6))
+  #     );
+  # @param os_argument_name [String] The string of the argument to check
+  # @param user_arguments [OSArgumentMap] Map of the arguments to check
+  def bad_os_typecasting(os_argument_name, user_arguments)
```

Comment:
@henryHorsey check out this section. I'm only typecasting the arguments that exist. If there is no value, then it doesn't show up. Extend this to get the defaults if you want them downstream.